posts/2020/06/nixos-rpi-wifi-router.md

```diff
@@ -0,0 +1,537 @@
+`nixos-sd-image-19.09.2435.9642f121eb1-aarch64-linux.img`. We'll assume this is in
+your `~/Downloads` folder.
+
+**Second**, figure out what device your SD card is. If you just plugged it in, you
+can determine this by looking at the end of the output of the `dmesg` command.
+For example,
+
+```sh
+$ dmesg
+[ 4591.053095] usb 3-10: new high-speed USB device number 5 using xhci_hcd
+[ 4591.201911] usb 3-10: New USB device found, idVendor=14cd, idProduct=168a, bcdDevice= 0.01
+[ 4591.201915] usb 3-10: New USB device strings: Mfr=1, Product=3, SerialNumber=2
+[ 4591.201917] usb 3-10: Product: USB Mass Storage Device
+[ 4591.201919] usb 3-10: Manufacturer: USB Device
+[ 4591.201921] usb 3-10: SerialNumber: 816820130806
+[ 4591.202955] usb-storage 3-10:1.0: USB Mass Storage device detected
+[ 4591.203140] scsi host10: usb-storage 3-10:1.0
+[ 4592.205933] scsi 10:0:0:0: Direct-Access     USB Mass  Storage Device  1.00 PQ: 0 ANSI: 0
+[ 4592.206338] sd 10:0:0:0: Attached scsi generic sg1 type 0
+[ 4592.207288] sd 10:0:0:0: [sdb] 31116288 512-byte logical blocks: (15.9 GB/14.8 GiB)
+[ 4592.207421] sd 10:0:0:0: [sdb] Write Protect is off
+[ 4592.207425] sd 10:0:0:0: [sdb] Mode Sense: 03 00 00 00
+[ 4592.207561] sd 10:0:0:0: [sdb] No Caching mode page found
+[ 4592.207567] sd 10:0:0:0: [sdb] Assuming drive cache: write through
+[ 4592.212993]  sdb: sdb1 sdb2
+[ 4592.214220] sd 10:0:0:0: [sdb] Attached SCSI removable disk
+```
+
+This let's us know that the SD card is the `/dev/sdb` device and has two partitions.
+Yours might be called `/dev/sdc` or something similar. It also might have more than
+two partitions. That is totally normal at this step.
+
+**Third**, we need to copy the NixOS image over to the SD card. We'll do this with
+the `dd` command. The SD card should *not* be mounted right now. Run the following
+command with the path to the image and the SD card device replaced as appropriate.
+
+```sh
+$ sudo dd if=~/Downloads/nixos-sd-image-19.09.2435.9642f121eb1-aarch64-linux.img of=/dev/sdb
+```
+
+Great! At this point, we have now flashed the SD card with our new NixOS!
+
+**Fourth**, it will save us a lot of typing if we copy over an existing
+configuration file to the SD card. Copy the text of the
+`/etc/nixos/configuration.nix` file at the top of this article to a new file,
+let's call it `~/Downloads/config.nix`. Fill in this file with the SSID and
+password that you want your network to have. Then, run the following commands
+to copy the configuration file to the SD card. Again, modify the paths here
+as needed.
+
+```sh
+$ mkdir -p ~/mount
+$ sudo mount /dev/sdb2 ~/mount
+$ sudo mkdir -p ~/mount/etc/nixos
+$ sudo cp ~/Downloads/config.nix ~/mount/etc/nixos/configuration.nix
+$ sudo umount ~/mount
+```
+
+**Fifth**, now unplug the SD card from your main machine, plug it into the
+Raspberry Pi! Attach the ethernet, keyboard, monitor, and power supply to the
+Pi you will be booting up into your first NixOS! :tada:
```

Comment:
This didn't turn into an emoji, plus lower down `:copyright:`, `:wave:` have the same problem. Some unicode symbols are recognized by Nikola (copyright, registered), the image ones are best handled by going to Emojipedia (e.g. https://emojipedia.org/waving-hand/), using the "copy" button there, and pasting into VSCode.